### PR TITLE
[SYCLomatic] Add rule for error check macro migration

### PR DESCRIPTION
**Pull request by intwanghao**

### CODE REVIEW

File: clang/tools/dpct/DpctOptRules/CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,7 @@
 set(dpct_opt_rule_files
   ${CMAKE_SOURCE_DIR}/../clang/tools/dpct/DpctOptRules/forceinline.yaml
   ${CMAKE_SOURCE_DIR}/../clang/tools/dpct/DpctOptRules/intel_specific_math.yaml
+  ${CMAKE_SOURCE_DIR}/../clang/tools/dpct/DpctOptRules/macro.yaml
```
Comment (AndyCHHuang) [inline clang/tools/dpct/DpctOptRules/CMakeLists.txt]: need a more specific file name like:
macro_checks.yaml
or
macro_cuda_checks.yaml

**Review by AndyCHHuang — APPROVED** [review]
LGTM

File: clang/tools/dpct/DpctOptRules/macro_checks.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Copyright (C) Intel Corporation
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+# See https://llvm.org/LICENSE.txt for license information.
+
+# This file is an example of user defined migration rules.
+# Applying this rule will make DPCT migrate macro CUDA_CHECK to DPCT_CHECK_ERROR.
+# This file will be installed to folder:
```
Comment (zhimingwang36) [inline clang/tools/dpct/DpctOptRules/macro_checks.yaml]: remove line 7 and 8

File: clang/tools/dpct/DpctOptRules/macro_checks.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+# {dpct_install_folder}/extensions/opt_rules.
+# The rule specified in this file can be imported with commandline option:
+# --rule-file={dpct_install_folder}/extensions/opt_rules/macro_checks.yaml
+---
+- Rule: rule_error_check_macro    # specify an unique name(ID) of the rule
```
Comment (zhimingwang36) [inline clang/tools/dpct/DpctOptRules/macro_checks.yaml]:
```suggestion
- Rule: rule_error_check_macro    # Specify a unique name (ID) of the rule
```

File: clang/tools/dpct/DpctOptRules/macro_checks.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  Kind: Macro                     # specify the kind of the rule
```
Comment (zhimingwang36) [inline clang/tools/dpct/DpctOptRules/macro_checks.yaml]:
```suggestion
  Kind: Macro                     # Specify the kind of the rule
```

File: clang/tools/dpct/DpctOptRules/macro_checks.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  Priority: Takeover              # select the priority if the rule conflicts with another one. value = Fallback/Default/Takeover
+  In: CUDA_CHECK                  # the target macro need to be migrate
```
Comment (zhimingwang36) [inline clang/tools/dpct/DpctOptRules/macro_checks.yaml]:
```suggestion
  In: CUDA_CHECK                  # the target macro needs to be migrated
```

File: clang/tools/dpct/DpctOptRules/macro_checks.yaml [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  Out: DPCT_CHECK_ERROR           # the new macro name
```
Comment (zhimingwang36) [inline clang/tools/dpct/DpctOptRules/macro_checks.yaml]:
```suggestion
  Out: DPCT_CHECK_ERROR           # the macro used in migration
```

File: clang/tools/dpct/DpctOptRules/macro_checks.yaml [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+  Includes: []                    # if the new macro depends on header files, specify it here as a list
```
Comment (zhimingwang36) [inline clang/tools/dpct/DpctOptRules/macro_checks.yaml]:
```suggestion
  Includes: []                    # if the macro used in Out field depends on header files, specify it here
```

File: clang/tools/dpct/DpctOptRules/macro_checks.yaml [hunk 8]
```diff
@@ -0,0 +1,18 @@
+# Copyright (C) Intel Corporation
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+# See https://llvm.org/LICENSE.txt for license information.
+
+# This file is an example of user defined migration rules.
+# Applying this rule will make DPCT migrate macro CUDA_CHECK to DPCT_CHECK_ERROR.
```
Comment (zhimingwang36) [inline clang/tools/dpct/DpctOptRules/macro_checks.yaml]:
```suggestion
# This rule target to migrate macro CUDA_CHECK to DPCT_CHECK_ERROR.
```

File: clang/tools/dpct/DpctOptRules/macro_checks.yaml [hunk 9]
```diff
@@ -0,0 +1,18 @@
+# Copyright (C) Intel Corporation
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+# See https://llvm.org/LICENSE.txt for license information.
+
+# This file is an example of user defined migration rules.
```
Comment (zhimingwang36) [inline clang/tools/dpct/DpctOptRules/macro_checks.yaml]:
```suggestion

```

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**